cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConfiguration.java

```diff
@@ -50,4 +62,103 @@ private KafkaConfiguration(String configuration, List<String> forbiddenOptions)
     public static KafkaConfiguration unvalidated(String string) {
         return new KafkaConfiguration(string, emptyList());
     }
+
+    /**
+     * Validate the configs in this KafkaConfiguration returning a list of errors.
+     * @param kafkaVersion The broker version.
+     * @return A list of error messages.
+     */
+    public List<String> validate(KafkaVersion kafkaVersion) {
+        List<String> errors = new ArrayList<>();
+        Map<String, ConfigModel> models = readConfigModel(kafkaVersion);
+        for (Map.Entry<String, String> entry: asOrderedProperties().asMap().entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            ConfigModel config = models.get(key);
+            if (config == null) {
+                //throw new RuntimeException("Unknown config " + key);
```

Comment:
Do we want to keep this?

Reply:
As far as Kafka itself is concerned it's not an error to include configs which are not known to the broker. The originals are available to plugins (e.g. Authorizer, or policy plugins) which extend `Configurable` and can thus be configured using custom parameters.

Right now we don't really have a way for users to add their own plugins (and that's complicated because they ought to be baked into the Kafka image), so in effect we know there are no plugins being used (at least by people using the strimzi images) and we could therefore error if an unknown config was given by the user.

So I guess it depends on whether and how we might start adding support for user-supplied plugins.

Reply:
TBH I would not mind if you just let these options pass through. But the commented out throw call looks strange there. I think you should replace it with comment explaining that passing unknown options is not necessarily a problem.